tests/st/policy/test_felix_gateway.py

```diff
@@ -0,0 +1,510 @@
+        }
+        self._apply_resources(host_endpoint_data, self.gateway)
+
+    @staticmethod
+    def _apply_resources(resources, host):
+        # Apply new resources.
+        host.writefile("resources",
+                       yaml.dump(resources, default_flow_style=False))
+        host.calicoctl("apply -f resources")
+
+    def assert_host_can_curl_ext(self):
+        try:
+            self.host.execute("curl --fail -m 1 -o /tmp/nginx-index.html %s" % self.ext_server_ip)
+        except subprocess.CalledProcessError:
+            _log.exception("Internal host failed to curl external server IP: %s",
+                           self.ext_server_ip)
+            self.fail("Internal host failed to curl external server IP: %s" % self.ext_server_ip)
+
+    def assert_host_can_not_curl_ext(self):
+        try:
+            self.host.execute("curl --fail -m 1 -o /tmp/nginx-index.html %s" % self.ext_server_ip)
+        except subprocess.CalledProcessError:
+            return
+        else:
+            self.fail("Internal host can curl external server IP: %s" % self.ext_server_ip)
+
+    def remove_pol_and_endpoints(self):
+        self.delete_all("pol")
+        self.delete_all("hostEndpoint")
+        # Wait for felix to remove the policy and allow traffic through the gateway.
+        retry_until_success(self.assert_host_can_curl_ext)
+
+    def delete_all(self, resource):
+        # Grab all objects of a resource type
+        objects = yaml.load(self.hosts[0].calicoctl("get %s -o yaml" % resource))
+        # and delete them (if there are any)
+        if len(objects) > 0:
+            self._delete_data(objects, self.hosts[0])
+
+    def _delete_data(self, data, host):
+        _log.debug("Deleting data with calicoctl: %s", data)
+        self._exec_calicoctl("delete", data, host)
+
+    @staticmethod
+    def _exec_calicoctl(action, data, host):
+        # use calicoctl with data
+        host.writefile("new_profile",
+                       yaml.dump(data, default_flow_style=False))
+        host.calicoctl("%s -f new_profile" % action)
+
+    @classmethod
+    def get_container_ip(cls, container_name):
+        ip = log_and_run(
+            "docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' %s" %
+            container_name)
+        return ip.strip()
+
+
+def wipe_etcd():
+    _log.debug("Wiping etcd")
+    # Delete /calico if it exists. This ensures each test has an empty data
+    # store at start of day.
+    curl_etcd(get_ip(), "calico", options=["-XDELETE"])
+
+    # Disable Usage Reporting to usage.projectcalico.org
+    # We want to avoid polluting analytics data with unit test noise
+    curl_etcd(get_ip(),
+              "calico/v1/config/UsageReportingEnabled",
+              options=["-XPUT -d value=False"])
+    curl_etcd(get_ip(),
+              "calico/v1/config/LogSeverityScreen",
+              options=["-XPUT -d value=debug"])
+
+
+def curl_etcd(ip, path, options=None, recursive=True):
```

Comment:
We should move this into the FV libraries. Not necessarily something you should do for this PR, but someone, soon.